src/containers/Board/SymbolDetails/SymbolDetails.js

```diff
@@ -15,55 +16,78 @@ export class SymbolDetails extends Component {
   constructor(props) {
     super(props);
 
-    const { type, label, text, img, boardId } = props.symbol;
+    this.defaultSymbol = { type: 'symbol', label: '', text: '', img: '', boardId: '' }
 
     this.state = {
-      symbol: {
-        type,
-        label,
-        text,
-        img,
-        boardId
-      }
+      symbol: this.defaultSymbol,
+      editingSymbols: props.editingSymbols,
+      activeStep: 0
     };
   }
 
-  handleSubmit = symbol => {
-    const { onSubmit } = this.props;
-    onSubmit(symbol);
+  componentWillReceiveProps = props => {
+    this.setState({ editingSymbols: props.editingSymbols })
+  }
+
+  editingSymbol = () => this.state.editingSymbols[this.state.activeStep]
+
+  currentSymbolProp = prop => {
+    const currentSymbol = this.editingSymbol()
+    return currentSymbol ? currentSymbol[prop] : this.state.symbol[prop]
+  }
+
+  updateEditingSymbol = (id, property, value) => state => {
+    const editingSymbols = state.editingSymbols.map( s =>
+      s.id === id ? {...s, ...{ [property]: value} } : s
+    )
+    return {...state, editingSymbols}
+  }
+
+  updateNewSymbol = (property, value) => state => {
```

Comment:
used the functional setState so it's possible to update 2 fields without one overwriting the other